samples/snippets/csharp/tutorials/inheritance/shape.cs

```diff
@@ -0,0 +1,169 @@
+// <Snippet1>
+using System;
+
+public abstract class Shape
+{
+   public abstract double Area { get; }
+   
+   public abstract double Perimeter { get; }
+ 
+   public override string ToString()
+   {
+      return GetType().Name;
+   }
+   public static double GetArea(Shape shape)
+   {
+      return shape.Area;
+   }
+
+   public static double GetPerimeter(Shape shape)
+   {
+      return shape.Perimeter;
+   }
+}
+// </Snippet1>
+
+namespace DerivedClasses
+{
+// <Snippet2>
+using System;
+
+public class Square : Shape
+{
+   double length; 
+   
+   public Square(double length)
+   {
+      this.length = length;
+   }
+
+   public double Side
```

Comment:
I'd prefer either read only auto implemented properties, or expression bodied members (for Area and Perimeter)